### PR TITLE
Word ladder and word ladder II

### LINKED ISSUE #685: add leetcode solution of 126 and 127
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

solutions for word ladder and word ladder 2

### Suggested Change

.

### Rationale

_No response_

### Urgency

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by mahek0620**
## Related Issue

Closes #685 

## Description

Added solution for leetcode problems 126 and 127 

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [X] Documentation update
- [ ] Other (specify): _______________

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+  - LeetCode
+  - Python
+  - Java
+  - C++
+description: "This is a solution to the word ladder II problem on LeetCode."
+---
+
+## Problem Description
+
+A transformation sequence from word beginWord to word endWord using a dictionary wordList is a sequence of words beginWord -> s1 -> s2 -> ... -> sk such that:
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:19]: **Bad Practice**

```md
-> s1 -> s2 -> ... -> sk
```

**Good Practice**

```md
$s1 -> s2 -> ... -> sk$
```

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+Every adjacent pair of words differs by a single letter.
+Every si for 1 <= i <= k is in wordList. Note that beginWord does not need to be in wordList.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:22]: **Bad Practice**

```md
1 <= i <= k
```

**Good Practice**

```md
$1 <= i <= k$
```

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+```
+Input: beginWord = "hit", endWord = "cog", wordList = ["hot","dot","dog","lot","log"]
+Output: []
+Explanation: The endWord "cog" is not in wordList, therefore there is no valid transformation sequence.
+```
+
+### Constraints
+
+- 1 <= beginWord.length <= 5
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:48]: replace `1 <= beginWord.length <= 5` to `$1 <= beginWord.length <= 5$`

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+- endWord.length == beginWord.length
+- 1 <= wordList.length <= 500
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:50]: replace `1 <= wordList.length <= 500` to `$1 <= wordList.length <= 500$`

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+```
+Input: beginWord = "hit", endWord = "cog", wordList = ["hot","dot","dog","lot","log"]
+Output: []
+Explanation: The endWord "cog" is not in wordList, therefore there is no valid transformation sequence.
+```
+
+### Constraints
+
+- 1 <= beginWord.length <= 5
+- endWord.length == beginWord.length
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:49]: replace `endWord.length == beginWord.length` to`$endWord.length == beginWord.length$`

File: dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+- 1 <= wordList.length <= 500
+- wordList[i].length == beginWord.length
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0126-word-ladder-II.md:51]: replace `wordList[i].length == beginWord.length` to `$wordList[i].length == beginWord.length$`

File: dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+  - C++
+description: "This is a solution to the word ladder problem on LeetCode."
+---
+
+## Problem Description
+
+A transformation sequence from word beginWord to word endWord using a dictionary wordList is a sequence of words beginWord -> s1 -> s2 -> ... -> sk such that:
+
+Every adjacent pair of words differs by a single letter.
+Every si for 1 <= i <= k is in wordList. Note that beginWord does not need to be in wordList.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md:22]: replace `1 <= i <= k` to `$1 <= i <= k$`

File: dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+  - LeetCode
+  - Python
+  - Java
+  - C++
+description: "This is a solution to the word ladder problem on LeetCode."
+---
+
+## Problem Description
+
+A transformation sequence from word beginWord to word endWord using a dictionary wordList is a sequence of words beginWord -> s1 -> s2 -> ... -> sk such that:
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md:19]: replace `-> s1 -> s2 -> ... ->` to `$-> s1 -> s2 -> ... ->$`

File: dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+**Example 2:**
+
+```
+Input: beginWord = "hit", endWord = "cog", wordList = ["hot","dot","dog","lot","log"]
+Output: 0
+Explanation: The endWord "cog" is not in wordList, therefore there is no valid transformation sequence.
+```
+
+### Constraints
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0127-word-ladder.md:44]: **Bad Practice**

```md
### Constraints

- 1 <= beginWord.length <= 10
- endWord.length == beginWord.length
- 1 <= wordList.length <= 5000
- wordList[i].length == beginWord.length
- beginWord, endWord, and wordList[i] consist of lowercase English letters.
- beginWord != endWord
- All the words in wordList are unique.
```

**Good Practice**

```md
### Constraints

- $1 <= beginWord.length <= 10$
- $endWord.length == beginWord.length$
- $1 <= wordList.length <= 5000$
- $wordList[i].length == beginWord.length$
- beginWord, endWord, and $wordList[i]$ consist of lowercase English letters.
- $beginWord != endWord$
- All the words in wordList are unique.
```

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+Input: ratings = [1,2,2]
+Output: 4
+Explanation: You can allocate to the first, second and third child with 1, 2, 1 candies respectively.
+The third child gets 1 candy because it satisfies the above two conditions.
+
+```
+Input: root = [1,2,2,null,3,null,3]
+Output: false
+```
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:47]: **Bad Practice**

```md
### Constraints

- n == ratings.length
- 1 <= n <= 2 * 10^4
- 0 <= ratings[i] <= 2 * 10^4
```

**Good Practice**

```md
### Constraints

- $n == ratings.length$
- $1 <= n <= 2 * 10^4$
- $0 <= ratings[i] <= 2 * 10^4$
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now update your code and resolve your docker and deployment issues and for Good practice

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+### Constraints
+
+- n == ratings.length
+- 1 <= n <= 2 * 10^4
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:51]: replace `1 <= n <= 2 * 10^4` to `$1 <= n <= 2 * 10^4$`

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+The third child gets 1 candy because it satisfies the above two conditions.
+
+```
+Input: root = [1,2,2,null,3,null,3]
+Output: false
+```
+
+### Constraints
+
+- n == ratings.length
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:50]: replace `n == ratings.length` to `$n == ratings.length$`

File: dsa-solutions/lc-solutions/0100-0199/0135-Candy.md [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+- 1 <= n <= 2 * 10^4
+- 0 <= ratings[i] <= 2 * 10^4
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0100-0199/0135-Candy.md:52]: replace `0 <= ratings[i] <= 2 * 10^4` to `$0 <= ratings[i] <= 2 * 10^4$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Update the code based on the suggestions, solved errors related to Docker, and handle deployment issues.

**Comment by mahek0620** [comment]
@Ajay-Dhangar the changes you suggested has been modified

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Comment by Ajay-Dhangar** [comment]
@mahek0620, to calculate the score for your leaderboard, you'll need to add something to those files and then try it out at the bottom. Below is the code you need to include for your author profile in the documentation:

```md
---

<h2>Authors:</h2>

<div style={{display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between', gap: '10px'}}>
{['github_username'].map(username => (
 <Author key={username} username={username} />
))}
</div>
```

**Pull request merged**